src/main/scala/io/smartdatalake/workflow/dataobject/DeltaLakeTableDataObject.scala

```diff
@@ -0,0 +1,150 @@
+  }
+
+  override def getDataFrame(partitionValues: Seq[PartitionValues] = Seq())(implicit session: SparkSession): DataFrame = {
+    val df = session.read.format("delta").load(hadoopPath.toString)
+    validateSchemaMin(df)
+    df
+  }
+
+  override def init(df: DataFrame, partitionValues: Seq[PartitionValues])(implicit session: SparkSession): Unit = {
+    // create tables if possible
+    prepare(session)
+    if (!isTableExisting) {
+      logger.info(s"($id) Creating table ${table.fullName}.")
+      writeDataFrame(df, createTableOnly = true, partitionValues)
+    }
+  }
+
+  override def writeDataFrame(df: DataFrame, partitionValues: Seq[PartitionValues])
+                             (implicit session: SparkSession): Unit = {
+    validateSchemaMin(df)
+    writeDataFrame(df, createTableOnly=false, partitionValues)
+  }
+
+  /**
+   * Writes DataFrame to HDFS/Parquet and creates Hive table.
+   * DataFrames are repartitioned in order not to write too many small files
+   * or only a few HDFS files that are too large.
+   */
+  def writeDataFrame(df: DataFrame, createTableOnly: Boolean, partitionValues: Seq[PartitionValues])
+                    (implicit session: SparkSession): Unit = {
+    val dfPrepared = if (createTableOnly) {
+      // create empty df with existing df's schema
+      session.createDataFrame(List.empty[Row].asJava, df.schema)
+    } else {
+      // pass DataFrame straight through if numInitialHdfsPartitions == -1, in this case the file size in the responsibility of the framework and must be controlled in custom transformations
+      if(numInitialHdfsPartitions == -1) df
+      // estimate number of partitions from existing data, otherwise use numInitialHdfsPartitions
+      else if (isTableExisting) {
+        val currentHdfsPath = HdfsUtil.prefixHadoopPath(HiveUtil.existingTickTockLocation(table.db.get, session, table.name), None)
+        HdfsUtil.repartitionForHdfsFileSize(df, currentHdfsPath.toString)
+      } else df.repartition(numInitialHdfsPartitions)
+    }
+
+    // write table and fix acls
+    if (partitions.isEmpty) dfPrepared.write.format("delta").save(hadoopPath.toString) else {
+      dfPrepared.write.format("delta").partitionBy(partitions:_*).save(hadoopPath.toString)
+    }
+    if (acl.isDefined) {
+      AclUtil.addACLs(acl.get, hadoopPath)(filesystem)
+    }
+    if (analyzeTableAfterWrite && !createTableOnly) {
+      logger.info(s"($id) Analyze table ${table.fullName}.")
+      HiveUtil.analyze(session, table.db.get, table.name, partitions, partitionValues)
+    }
+  }
+
+  override def isDbExisting(implicit session: SparkSession): Boolean = {
+    session.catalog.databaseExists(table.db.get)
+  }
+
+  override def isTableExisting(implicit session: SparkSession): Boolean = {
+    session.catalog.tableExists(table.db.get, table.name)
+  }
+
+  /**
+   * list hive table partitions
+   */
+  override def listPartitions(implicit session: SparkSession): Seq[PartitionValues] = {
```

Comment:
Check Hadoop/SparkFileDataObject to list partitions

Reply:
Done: https://github.com/smart-data-lake/smart-data-lake/pull/13/commits/7c147455d1d452fd9b69d429fbd18e54b73c4da7